src/sortStudents.ts

```diff
@@ -1,16 +1,64 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade'
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+const averageGrade = (student: Student): number => student.grades
+  .reduce((sum, grade) => sum + grade) / student.grades.length;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copie: Student[] = [...students];
+
+  switch (sortBy) {
+    case (SortType.Name):
+    case (SortType.Surname):
+      return copie.sort((studentA: Student, studentB: Student) => {
+        if (order === 'asc') {
+          return studentA[sortBy].localeCompare(studentB[sortBy]);
+        }
+
+        return studentB[sortBy].localeCompare(studentA[sortBy]);
+      });
```

Comment:
you can make like that
```suggestion
      return copie.sort((studentA: Student, studentB: Student) => order === 'asc'
        ? studentA[sortBy].localeCompare(studentB[sortBy])
        :  studentB[sortBy].localeCompare(studentA[sortBy])
       );
```